go/tools/builders/checker_main.go

```diff
@@ -44,39 +43,54 @@ import (
 // do not return an error so as not to unnecessarily interrupt builds.
 func run(args []string) error {
 	archiveFiles := multiFlag{}
+	importMapIn := multiFlag{}
+	filenames := multiFlag{}
 	flags := flag.NewFlagSet("checker", flag.ContinueOnError)
 	flags.Var(&archiveFiles, "archivefile", "Archive file of a direct dependency")
+	flags.Var(&importMapIn, "importmap", "Import maps of a direct dependency")
+	flags.Var(&filenames, "filename", "A source file being compiled")
+	vetTool := flags.String("vet_tool", "", "The vet tool")
+	stdLib := flags.String("stdlib", "", "The directory containing stdlib packages")
+	packageList := flags.String("package_list", "", "The file containing the list of standard library packages")
 	if err := flags.Parse(args); err != nil {
 		log.Println(err)
 		return nil
 	}
-	goroot, ok := os.LookupEnv("GOROOT")
-	if !ok {
-		log.Fatalf("GOROOT not set")
-	}
-	importsToArchives := make(map[string]string)
-	for _, a := range archiveFiles {
-		kv := strings.Split(a, "=")
-		if len(kv) != 2 {
-			continue // sanity check
+
+	if enableVet {
+		vcfgFile, err := makeVetCfg(*packageList, *stdLib, archiveFiles, importMapIn, filenames)
+		if err != nil {
+			log.Printf("error creating vet config: %v", err)
+			return nil
+		}
+		defer os.Remove(vcfgFile)
+		findings, err := runVet(*vetTool, vcfgFile)
+		if err != nil {
+			log.Printf("error running vet:\n%v\n", err)
```

Comment:
The idea is that only legitimate analysis findings should fail builds. Other internal errors (e.g. type-checking, loading) should merely emit a warning message. Developers shouldn't be blocked by errors they have no control over.

Reply:
I was thinking of the third kind of error. The vet wrapper is a hack that might break if the contract between vet and and the go build program changes in an upcoming version of Go. I don't think we should fail users if our implementation fails to handle some of these edge cases. OTOH, failing hard will ensure that these bugs are fixed in a timely manner.

Since users have to opt-in to use vet and can easily disable it, the impact on developer productivity will probably be minimal. I'll make all vet errors hard errors then.